src/App.tsx

```diff
@@ -1,73 +1,108 @@
+  let timerId = 0;
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  return (...args: any) => {
+    window.clearTimeout(timerId);
+    timerId = window.setTimeout(() => callback(...args), delay);
+  };
+};
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [inputFocused, setInputFocused] = useState(false);
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const applyQuery = useCallback(debounce(setAppliedQuery), []);
+
+  const handleSelection = (person: Person | null) => {
+    setSelectedPerson(person);
+
+    if (person) {
+      setQuery(person.name);
+      applyQuery(person.name);
+    }
+  };
+
+  const getFilteredPeople = useMemo(
+    () =>
+      [...peopleFromServer].filter(person =>
+        person.name.toLowerCase().includes(appliedQuery.toLowerCase()),
+      ),
+    [appliedQuery],
+  );
 
   return (
     <div className="container">
       <main className="section is-flex is-flex-direction-column">
         <h1 className="title" data-cy="title">
-          {`${name} (${born} - ${died})`}
+          {selectedPerson
+            ? `${selectedPerson.name} (${selectedPerson.born} - ${selectedPerson.died})`
+            : `No selected person`}
         </h1>
 
         <div className="dropdown is-active">
           <div className="dropdown-trigger">
             <input
               type="text"
+              value={query}
               placeholder="Enter a part of the name"
               className="input"
               data-cy="search-input"
+              onFocus={() => setInputFocused(true)}
+              onBlur={() => setInputFocused(false)}
+              onChange={e => {
+                const newValue = e.target.value;
+
+                if (selectedPerson && newValue !== selectedPerson.name) {
+                  setSelectedPerson(null);
+                }
+
+                setQuery(newValue);
+                applyQuery(newValue);
+              }}
             />
           </div>
 
           <div className="dropdown-menu" role="menu" data-cy="suggestions-list">
-            <div className="dropdown-content">
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Bernard Haverbeke</p>
+            {inputFocused && (
+              <div
+                className="dropdown-content"
+                onMouseDown={e => e.preventDefault()}
+              >
+                <Autocomplete
+                  delay={300}
+                  filteredPeople={getFilteredPeople}
+                  onSelected={handleSelection}
+                />
```

Comment:
Avoid generating keys during render. Consider using a stable identifier from the data, such as `person.id` or `person.slug`, to ensure keys are consistent across renders.